### PR TITLE
Add Support for private muc message

### PR DESCRIPTION
**Pull request by inkhey**
Fix issue https://github.com/jitsi/jitsi-meet/issues/2073, and add backend for feature https://github.com/jitsi/jitsi-meet/issues/2074.

### CODE REVIEW

File: modules/xmpp/ChatRoom.js [hunk 1]
```diff
@@ -689,6 +689,31 @@ export default class ChatRoom extends Listenable {
...
+            type: 'chat' });
+
+        msg.c('body', body).up();
+        if (nickname) {
+            msg.c('nick', { xmlns: 'http://jabber.org/protocol/nick' })
+                .t(nickname)
+                .up()
+                .up();
+        }
+        msg.c('x', { xmlns: 'http://jabber.org/protocol/muc#user' })
```
Comment (inkhey) [inline modules/xmpp/ChatRoom.js]: Look like this line isn't needed, for an unknown reason, this element is already send when using _send_ method below.

File: JitsiConferenceEvents.js [hunk 2]
```diff
@@ -124,6 +124,11 @@ export const LOCK_STATE_CHANGED = 'conference.lock_state_changed';
 export const MESSAGE_RECEIVED = 'conference.messageReceived';
 
 /**
+ * New private text message was received.
+ */
+export const MESSAGE_Private_RECEIVED = 'conference.messagePrivateReceived';
```
Comment (saghul) [inline JitsiConferenceEvents.js]: incorrect capitalization

File: modules/xmpp/ChatRoom.js [hunk 3]
```diff
@@ -689,6 +689,31 @@ export default class ChatRoom extends Listenable {
 
     /**
      *
+     * @param id
```
Comment (saghul) [inline modules/xmpp/ChatRoom.js]: document these

File: modules/xmpp/ChatRoom.js (hunk 1 above)
Comment (saghul) [reply modules/xmpp/ChatRoom.js]: If it's not needed... drop it then?

File: modules/xmpp/ChatRoom.js [hunk 4]
```diff
@@ -857,9 +882,16 @@ export default class ChatRoom extends Listenable {
         }
 
         if (txt) {
-            logger.log('chat', nick, txt);
-            this.eventEmitter.emit(XMPPEvents.MESSAGE_RECEIVED,
-                from, nick, txt, this.myroomjid, stamp);
+            if (type === 'chat'){
```
Comment (saghul) [inline modules/xmpp/ChatRoom.js]: style: space after )

File: service/xmpp/XMPPEvents.js [hunk 5]
```diff
@@ -97,6 +97,9 @@ const XMPPEvents = {
     // received.
     MESSAGE_RECEIVED: 'xmpp.message_received',
 
+    // Designates an event indicating that a private XMPP message in the MUC was
+    // received.
+    MESSAGE_PRIVATE_RECEIVED: 'xmpp.message_private_received',
```
Comment (saghul) [inline service/xmpp/XMPPEvents.js]: please rename this to PRIVATE_MESSAGE_RECEIVED. it reads much better

File: modules/xmpp/ChatRoom.js [hunk 6]
```diff
@@ -689,6 +689,31 @@ export default class ChatRoom extends Listenable {
 
     /**
      *
+     * @param id
+     * @param body
+     * @param nickname
+     */
+    sendPrivateMessage(id,body, nickname) {
```
Comment (saghul) [inline modules/xmpp/ChatRoom.js]: style: space after comma

**Comment by inkhey** [comment]
Done.

File: modules/xmpp/ChatRoom.js [hunk 7]
```diff
@@ -687,6 +687,30 @@ export default class ChatRoom extends Listenable {
         this.eventEmitter.emit(XMPPEvents.SENDING_CHAT_MESSAGE, body);
     }
 
+    /**
+     *
```
Comment (saghul) [inline modules/xmpp/ChatRoom.js]: please complete the docs here

File: JitsiConference.js [hunk 8]
```diff
@@ -567,6 +567,15 @@ JitsiConference.prototype.sendTextMessage = function(message) {
     }
 };
 
+/**
+ * Sends text message to the other participants in the conference
```
Comment (saghul) [inline JitsiConference.js]: please mention the message is private

**Comment by inkhey** [comment]
Done.
Anything else ?

**Comment by saghul** [comment]
Jenkins please test.

**Comment by saghul** [comment]
Jenkins, ok to test.

**Comment by saghul** [comment]
@inkhey can you please rebase? I think tests are failing because of that.

**Comment by inkhey** [comment]
Like this ?

**Comment by saghul** [comment]
@inkhey yep, thanks!

**Comment by saghul** [comment]
Jenkins, please test.

**Comment by saghul** [comment]
@inkhey Please run the linter when you commit:

~~~
/private/tmp/lib-jitsi-meet/JitsiConference.js
  593:1  error  Expected line before comment  lines-around-comment

/private/tmp/lib-jitsi-meet/modules/xmpp/ChatRoom.js
  704:32  error  Unexpected string concatenation      prefer-template
  704:44  error  Infix operators must be spaced       space-infix-ops
  704:45  error  Strings must use singlequote         quotes
  704:48  error  Infix operators must be spaced       space-infix-ops
  718:1   error  More than 2 blank lines not allowed  no-multiple-empty-lines

/private/tmp/lib-jitsi-meet/service/xmpp/XMPPEvents.js
  103:5  error  Expected line before comment                    lines-around-comment
  176:1  error  Line 176 exceeds the maximum line length of 80  max-len
  176:5  error  Expected line before comment                    lines-around-comment
  179:5  error  Expected line before comment                    lines-around-comment

✖ 10 problems (10 errors, 0 warnings)
~~~

You can do that manually with `npm run lint`, but it should run automatically on your local machine when you commit 😮 .

**Comment by inkhey** [comment]
Sorry for the delay :-) .

**Comment by saghul** [comment]
Tested in https://github.com/jitsi/lib-jitsi-meet/pull/703 and landed, cheers @inkhey !

**Pull request closed without merging**